src/diagnostics.rs

```diff
@@ -271,7 +271,9 @@ pub fn report_error<'tcx, 'mir>(
                     helps.push((None, format!("see https://doc.rust-lang.org/nightly/std/sync/atomic/index.html#memory-model-for-atomic-accesses for more information about the Rust memory model")));
                 }
                 if *retag_explain {
-                    helps.push((None, "retags permit optimizations that insert speculative (or loop-hoisted) reads/writes. Therefore from the perspective of data races, a retag has the same implications as a read/write.".to_owned()));
+                    helps.push((None, "retags occur on all (re)borrows and as well as when references are copied or moved. \
+                        Retags permit optimizations that insert speculative reads or writes. \
+                        Therefore from the perspective of data races, a retag has the same implications as a read or write.".to_owned()));
```

Comment:
```suggestion
                    helps.push((None, "retags permit optimizations that insert speculative reads or writes. \
                        Therefore, a retag has the same implications as a read or write from the perspective of data races.".to_owned()));
                    helps.push((None, "retags occur on all (re)borrows and as well as when references are copied or moved".to_owned()));
```

Reply:
Should the lines be split to one help message each? Why would `why is a retag?` be answered before `what is a retag?`?

Reply:
I was thinking we would start with "why is this a race" -- because of speculative reads/writes. Then we can explain where those come from. But you are right, the other order would also make sense.

Splitting into 3 messages also works for me, I just felt it was too long for 1.^^